### PR TITLE
Separate GUI sources from core sources

### PR DESCRIPTION
**Pull request by louib**
This PR splits the GUI source files from the core source files. The immediate goal is to allow the CLI to require only a minimum number of dynamic libraries. The long term goal is to create an architectural boundary around the core module, in preparation of [libkdbx](https://github.com/keepassxreboot/keepassxc/issues/5717).

The only real challenge in this PR was to deal with the network related modules. I ended up creating a `networking` directory and moving all the network related modules from `core/` in there.

In the future, I think we could move all the core code directly into the `core/` folder, and create a `CMakeLists.txt` specifically for that folder.

I tested mainly with `WITH_XC_ALL=ON` to make sure most of the cmake code was triggered.

Before the changes:
```
ldd src/keepassxc | wc -l
58
ldd src/cli/keepassxc-cli | wc -l
60
```
Basically, the CLI was using exactly the same dynamic libraries as the GUI, plus `libncursesw` and `libreadline`.

After the changes:
```
ldd src/keepassxc | wc -l
58
ldd src/cli/keepassxc-cli | wc -l
42
```

There's still a bunch of libraries that the CLI is linking against for reasons that I don't understand, like `libQt5Gui`. This might be included in `Qt5::Core`, but I'm not sure.

## Testing strategy
unit tests and launched the GUI and CLI apps as a smoke test.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Refactor (significant modification to existing code)

### CODE REVIEW

**Comment by droidmonkey** [comment]
This should be merged next after the 2.7.6 PR's

**Review by droidmonkey — APPROVED** [review]
